src/machine/events/HandleRunningPods.ts

```diff
@@ -59,14 +60,52 @@ export function handleRuningPods(): OnEvent<RunningPods.Subscription, NoParamete
         }
 
         if (deployGoal && desc) {
-            await updateGoal(context, deployGoal, {
-                state: SdmGoalState.success,
-                description: desc,
-                url: deployGoal.url,
-            });
+
+            // grab deploymentStarted event
+            const targetDeployment = await fetchDeploymentTarget(context, pod);
+
+            const numCurrentPods = pod.containers[0].image.pods.filter(deployedPod => {
+                return pod.environment = deployedPod.environment;
+            }).length;
+            const numTargetPods = targetDeployment.targetReplicas;
+            desc = desc + ` (${numCurrentPods}/${numTargetPods}`;
+            if (numCurrentPods === numTargetPods) {
+                // then we know we have a successful deployment
+                // need to find commits between current and previous!
+                await updateGoal(context, deployGoal, {
+                    state: SdmGoalState.success,
+                    description: desc,
+                    url: deployGoal.url,
+                });
+            } else {
+                await updateGoal(context, deployGoal, {
+                    state: SdmGoalState.in_process,
+                    description: desc,
```

Comment:
I really like that the description continues to update so you can see the progress.  It will be cool to see how this looks in action